src/events/balancer.py

```diff
@@ -169,6 +194,12 @@ def _on_config_changed(self, _: EventBase) -> None:
 
     def rebalance(self, event: ActionEvent) -> None:
         """Handles the `rebalance` Juju Action."""
+        if self.charm.state.runs_broker:
+            available_brokers = [broker.unit_id for broker in self.charm.state.brokers]
+        else:
+            brokers: list = self.charm.state.balancer.broker_capacities.get("brokerCapacities", [])
```

Comment:
**todo:** I don't think this is the right way to get units.
I may be wrong, but I **believe** that they should be in the relation data (which we abstract a bit with state, admittedly).

Check the keys of `self.charm.state.balancer.relation.data` or maybe `self.charm.state.balancer.relation.units`, I think it's in one of those.

Reply:
Come to think of it, it might be clever to actually instantiate a `brokers: list[KafkaBroker]` as a property inside `PeerCluster`. Maybe too complicated for now, but seems like a good idea for when we do support large-large-deployments.

Reply:
`.units` was indeed the answer, changed in ff03984bc8cd

The `brokers` property is a good idea, I'll keep it in mind in the future